src/diskusage.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2015 Jolla Ltd.
+ * Contact: Thomas Perl <thomas.perl@jolla.com>
+ *
+ * You may use this file under the terms of the BSD license as follows:
+ *
+ * "Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Nemo Mobile nor the names of its contributors
+ *     may be used to endorse or promote products derived from this
+ *     software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE."
+ */
+
+#ifndef DISKUSAGE_H
+#define DISKUSAGE_H
+
+#include <QObject>
+#include <QVariant>
+#include <QJSValue>
+#include <QScopedPointer>
+
+class DiskUsagePrivate;
+
+class DiskUsage : public QObject
+{
+    Q_OBJECT
+    Q_DECLARE_PRIVATE(DiskUsage)
+
+    // True while calculation takes place
+    Q_PROPERTY(bool working
+            READ working
+            NOTIFY workingChanged)
+
+public:
+    explicit DiskUsage(QObject *parent=0);
+    virtual ~DiskUsage();
+
+    // Calculate the disk usage of the given paths, then call
+    // callback with a QVariantMap (mapping paths to usages in bytes)
+    Q_INVOKABLE void calculate(const QStringList &paths, QJSValue callback);
```

Comment:
It might be fun for the user to watch the disk usage being counted for each item (as opposed to waiting until the whole thing is done). That would imply separate context for each item. Anyway, it's up to you.